dashboard/src/actions/repos.ts

```diff
@@ -175,6 +205,36 @@ export const installRepo = (
   };
 };
 
+export const updateRepo = (
+  name: string,
+  namespace: string,
+  repoURL: string,
+  authHeader: string,
+  customCA: string,
+  syncJobPodTemplate: string,
+): ThunkAction<Promise<boolean>, IStoreState, null, AppReposAction> => {
+  return async (dispatch, getState) => {
+    try {
+      const syncJobPodTemplateObj = parsePodTemplate(syncJobPodTemplate);
+      const ns = getTargetNS(getState, namespace);
+      dispatch(requestRepoUpdate());
+      const data = await AppRepository.update(
+        name,
+        ns,
+        repoURL,
+        authHeader,
+        customCA,
+        syncJobPodTemplateObj,
```

Comment:
Like this - when updating the rep, we just pass the relevant data to the backend. Similarly, we could have a getRepo which returns the relevant data for the frontend (or other consumers) without them needing to separately load secrets etc.